### PR TITLE
feat: add new linea mainnet network

### PR DESCRIPTION
**Pull request by VGau**
## Explanation

- Added new Linea mainnet network as a default network.
- Updated `@metamask/eth-json-rpc-infura` package version to 8.1.0 to support linea networks
- ⚠️ Linea mainnet network won't be display on MM before the release date. This network is under feature toggle for now.

See: https://github.com/ConsenSys/zkevm-monorepo/issues/551

### Before
Screenshot 2023-06-05 at 1 16 41 PM

### After
Screenshot 2023-06-05 at 1 51 30 PM
Screenshot 2023-06-05 at 1 51 49 PM

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: app/scripts/controllers/network/create-network-client.ts
Comment (Gudahtt) [inline app/scripts/controllers/network/create-network-client.ts:62]: Is this still needed? If so, when is this planned to be removed?

File: shared/constants/network.ts
Comment (Gudahtt) [inline shared/constants/network.ts:320]: This site does not seem to be working

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js
Comment (Gudahtt) [inline ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:104]: Why was this added condition needed here?

File: ui/helpers/utils/i18n-helper.tsx
Comment (Gudahtt) [inline ui/helpers/utils/i18n-helper.tsx:170]: Why do these two networks need a different "label key" than the others?

File: app/scripts/controllers/network/create-network-client.ts
Comment (VGau) [reply app/scripts/controllers/network/create-network-client.ts:62]: Yes it is needed because we use the testnet RPC url for now. We will update this url once the linea mainnet network is ready.

File: shared/constants/network.ts
Comment (VGau) [reply shared/constants/network.ts:320]: Yes this url is not working for the moment but will be the one for linea mainnet :)

File: ui/helpers/utils/i18n-helper.tsx
Comment (VGau) [reply ui/helpers/utils/i18n-helper.tsx:170]: It is because `app/_locales` `messages.json` files, use networks name as keys. `linea-goerli` and `linea-mainnet` networks have this `-` character in their name . `app/_locales` `message.json` files cannot have keys containing `-` character.
That's why I use `lineaGoerli` and `lineaMainnet` as keys in `app/_locales` files and that's why I need this `getNetworkLabelKey` function to map real network names with `app/_locales` `message.json` files keys.

File: ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js
Comment (VGau) [reply ui/pages/settings/networks-tab/networks-list-item/networks-list-item.js:104]: It is a temporay condition because linea mainnet network use the linea goerli rpc url and chainId

File: shared/modules/network.utils.ts
Comment (danjm) [inline shared/modules/network.utils.ts:61]: Nit: if `currentDateHigherThanReleaseDate` is false, then we can return false before we make the `jsonRpcRequest`

File: ui/components/multichain/network-list-menu/network-list-menu.js
Comment (danjm) [inline ui/components/multichain/network-list-menu/network-list-menu.js:66]: This will result in a network request every time the network menu is rendered. We could instead call `shouldShowLineaMainnet` in the constructor of the preferences controller, and store the result in preferences controller state, so that once mainnet is released, further requests are not needed.

File: app/scripts/controllers/network/create-network-client.ts
Comment (Gudahtt) [reply app/scripts/controllers/network/create-network-client.ts:62]: Hiding the Linea mainnet network prior to release makes sense, but I don't see the purpose in making it act like Linea Goerli. If we do this, then it will require another code change to make the network work properly. If that's so, then why bother adding it all until that point?

Comment (VGau) [reply app/scripts/controllers/network/create-network-client.ts:62]: Indeed :) The idea was to put the linea-goerli rpc url for the linea mainnet to test the integration but I think it is better to put the real rpc url. I will change that thanks :)

File: ui/helpers/utils/i18n-helper.tsx
Comment (Gudahtt) [reply ui/helpers/utils/i18n-helper.tsx:170]: Thanks, makes sense.

This does go against a guideline we have about using variables for the `t` function - we prefer to use string literals instead so that we can more effectively find unused localized messages and remove them. But this was a pre-existing pattern in this case and you're just extending it, so this seems fine for this PR

File: ui/components/multichain/network-list-menu/network-list-menu.js
Comment (Gudahtt) [reply ui/components/multichain/network-list-menu/network-list-menu.js:66]: Great point. Do we need the `eth_chainId` call for this? It would be ideal to avoid network call side-effects if possible, and just use the timer.

File: ui/helpers/utils/i18n-helper.tsx
Comment (VGau) [reply ui/helpers/utils/i18n-helper.tsx:170]: I agree that is not a very clean solution but it is the only solution I found. If you have another suggestion feel free to share it  😄

Comment (Gudahtt) [reply ui/helpers/utils/i18n-helper.tsx:170]: Our solution elsewhere is to create a mapping of keys (e.g. network name) to localized _messages_ (not message keys), e.g. as we've done here: https://github.com/MetaMask/metamask-extension/blob/develop/shared/notifications/index.js

While I would prefer that solution, I'm actually glad that you haven't done it here because it would increase the scope of the PR too much. I'm just mentioning it here for future reference.

File: shared/modules/network.utils.ts
Comment (Gudahtt) [inline shared/modules/network.utils.ts:71]: What is the purpose of this chain ID check?

Comment (VGau) [reply shared/modules/network.utils.ts:71]: It just to check if the linea mainnet rpc url is working. We can choose another endpoint if you want :)

File: ui/components/multichain/network-list-menu/network-list-menu.js
Comment (danjm) [reply ui/components/multichain/network-list-menu/network-list-menu.js]: We do need a network call, because they don't want to show the network before infura's Linea mainnet is actually live. So the logic is supposed to be: don't show the network before this date AND don't show it before infura's mainnet is live. The date based condition ensures that the network won't be shown before the planned launch date, the network call check allows the launch date to be moved.

**Review by danjm — APPROVED** [review]
This looks good to me

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Looks good apart from the issue with the network call; I can approve after that is resolved

File: shared/modules/network.utils.ts
Comment (danjm) [reply shared/modules/network.utils.ts:71]: good point

**Review by danjm — CHANGES_REQUESTED** [review]
removing my approval because of the network call issue

File: shared/modules/network.utils.ts
Comment (danjm) [reply shared/modules/network.utils.ts:71]: I think there might be a product decision to make here though. Effectively the network would be hidden at launch for anyone not on an infura network. How many people do we think that is, and is that number acceptable given the goals of the network launch?

Comment (danjm) [reply shared/modules/network.utils.ts:71]: There is an alternative, that would require product approval, where we should the network after the date and only make a network request once the user attempts to switch to the linea network, and if it fails we could show a custom error message (e.g. "Thanks for your interest in Linea Mainnet, it will be available soon!"), or hide the network from the list or just fallback to our current failure mode when failing to connect to a network (loading spinner and then the option to try again or switch networks)

Comment (Gudahtt) [reply shared/modules/network.utils.ts:71]: That seems like a good idea to me!

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by danjm — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**
